docs/groovy/how-to-guides/query-string-overview.md

```diff
@@ -0,0 +1,211 @@
+
+The following code block provides an example of a simple query string that creates a new column called `NewColumn` in a table called `source`. The string passed into [`update`](../reference/table-operations/select/update.md) is the query string. The query string `NewColumn = 1` defines a [formula](./formulas.md), where the left-hand side (LHS) is the name of the column to be created and the right-hand side (RHS) defines the values that will be assigned to that column. In this case, each value in `NewColumn` is 1.
+
+```groovy order=source
+source = emptyTable(10).update("NewColumn = 1")
+```
+
+## Query string examples
+
+A query string defines a [formula](./formulas.md) or a [filter](./filters.md):
+
+- [Formulas](./formulas.md) contain assignments.
+- [Filters](./filters.md) evaluate to True/False.
+
+Query strings can contain a wide array of expressions. Different types of expressions are explored via examples in the following subsections.
+
+### Formulas
+
+Formulas contain assignments. These assignments usually come in the form of `LHS = RHS`, where the left-hand-side (`LHS`) contains the name of the new column that will be created, and the right-hand-side (`RHS`) defines the values that will be assigned to that column.
+
+The following example creates a table with five columns. Each column is created via a formula:
+
+```groovy test-set=1 order=source
+source = emptyTable(20).update(
+    "X = 0.2 * ii",
+    "Y = sin(X)",
+    "Z = (ii % 2 == 0) ? `Even` : `Odd`",
+    "RandomNumber = randomDouble(-1, 1)",
+    "Timestamp = now() + ii * SECOND",
+)
+```
+
+The formulas used above all leverage query language built-ins including [built-in constants](./built-in-constants.md), [built-in variables](./built-in-variables.md), and [built-in functions](./built-in-functions.md).
+
+### Filters
+
+Filters evaluate to True/False and are used to determine which rows are included in the result table. Consider the following example, which uses the table created in the [previous section](#formulas):
+
+```groovy test-set=1 order=resultLessThan,resultEven,resultRng,resultConjunctive,resultDisjunctive
+resultLessThan = source.where("X <= 1.2")
+resultEven = source.where("Z == `Even`")
+resultRng = source.where("RandomNumber >= 0.21")
+resultConjunctive = source.where("X <= 2.4", "Z == `Even`")
+resultDisjunctive = source.where("X > 1.6 || Z == `Even`")
+```
+
+For more on filtering table data, see [Use filters](./filters.md).
+
+### Literals
+
+A literal is a fixed value that's used in a query string. The [previous section](#filters) used numeric literals in some of the filter statements. The value is defined in the query string itself, not outside of it and then passed in as a parameter. The following example creates a table with several columns, each containing a literal value. The table metadata is included to show the resultant column data types.
```

Comment:
Probably worth making the same change in the Python version as well. String literals also get used. Either this, or you can specifically mention numeric and string literals, adding a link to each page. My suggestion is the quick route, and what I'd probably do since the three literal guides are linked to at the bottom of the section.

```suggestion
A literal is a fixed value that's used in a query string. The [previous section](#filters) used literals in some of the filter statements. The value is defined in the query string itself, not outside of it and then passed in as a parameter. The following example creates a table with several columns, each containing a literal value. The table metadata is included to show the resultant column data types.
```